rskj-core/src/main/java/co/rsk/peg/BridgeSerializationUtils.java

```diff
@@ -273,7 +273,7 @@ private static Federation deserializeFederationWithDeserializer(
             federationMembers.add(member);
         }
 
-        return new Federation(federationMembers, creationTime, creationBlockNumber, networkParameters);
+        return new StandardMultisigFederation(federationMembers, creationTime, creationBlockNumber, networkParameters);
```

Comment:
Now this method changes. The way I see it, it's serializing a StandardMultisigFederation, not any  federation type. So I'd change the signature to receive an object of `StandardMultisigFederation` type

Reply:
Just in case, check where it's used. Make sure that it's always called with a SatandarMultisigFederation, if it's not we might need to reconsider. And I would also change the name of the method to make it clear that it is not serializing any federation

Reply:
I think so because the methods `deserializeErpFederation` and `deserializeP2shErpFederation` exist too.

Reply:
Should we rename it then to `deserializeStandardMultisigFederationWithDeserializer` and change the signature to return a `StandardMultisigFederation`?

Reply:
yes good catch. should we rename all the "deserialize / serialize Federation..." to "deserialize / serialize StandardMultisigFederation" also?

Reply:
If that's what they do, yes

Reply:
I would also change the return type of this function, to return a StandarMultisigFederation. You are expecting exactly that, not any generic Federation

Reply:
that breaks some things so i should take a better look at it

Reply:
for example, this code inside testGetNewFederationPostMultiKey function from BridgeStorageProviderTest throws an exception:

```
        try (MockedStatic<BridgeSerializationUtils> bridgeSerializationUtilsMocked = mockStatic(BridgeSerializationUtils.class)) {
            bridgeSerializationUtilsMocked.when(() -> BridgeSerializationUtils.deserializeStandardMultisigFederation(
                any(byte[].class),
                any(NetworkParameters.class))
            ).then((InvocationOnMock invocation) -> {
                deserializeCalls.add(0);
                byte[] data = invocation.getArgument(0);
                NetworkParameters networkParameters = invocation.getArgument(1);
                // Make sure we're deserializing what just came from the repo with the correct BTC context
                assertArrayEquals(new byte[]{(byte) 0xaa}, data);
                assertEquals(networkParameters, bridgeConstants.getBtcParams());
                return federation;
            });
```

because the _deserializeStandardMultisigFederation_ returns a StandardMultisigFederation, that cannot be cast to a ErpFederation

Reply:
we may reconsider changing the tests